### DISCUSSION TITLE
Using montage with a .png or .tga produces a very light output

### DISCUSSION THREAD

**Discussion by Varunda** [ImageMagick/ImageMagick#3413] [Help] [answered]
When combining 4 .tga in a 2x2 square with a .tga output, the output is much lighter than the input. 

Example input: https://i.imgur.com/FOBloOX.png

Example output: https://i.imgur.com/yymI0nF.png

Command: `magick montage *[0-3]_color.tga -geometry 1024x1024+0+ out.png`

The output is much lighter than expected, with the input being the top left of the output. I've tried using both .tga and .png files for the input, and outputting both .png and .tga files, and the result is always the same.

I've tried two different versions of IM:

```
Version: ImageMagick 7.0.11-3 Q8 x64 2021-03-06 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192829337
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zip zlib
```

```
Version: ImageMagick 7.0.11-3 Q16 x64 2021-03-06 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192829337
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zip zlib
```

Both versions create visually identical outputs, with the same unwanted lightness on both.

Any direction to how I could fix this problem would be fantastic

**Comment by fmw42** [comment]
Please post one TGA file so we can see if it happens when converting tga to png? Have you tried that?

**Reply by Varunda** [reply]
When converting a .tga to a .png, the same lightness appears. I've attached a zip with one of the .tgas and the output it produces when using `magick in.tga out.png`
[Indar_-16_12.0_color.zip](https://github.com/ImageMagick/ImageMagick/files/6160240/Indar_-16_12.0_color.zip)

**Comment by snibgo** [comment]
With your input file, and IM v7.0.8-64, I can't reproduce the problem.

> The output is much lighter than expected...

This suggests a linear RGB versus non-linear sRGB problem. Perhaps your input file has been mangled by the imgur website. Please put your files in a zip file, upload it to somewhere, and post the link.

**Reply by fmw42** [reply]
Snibgo, his input that he provide is PNG not his original TGA. So we need his actual TGA. To post that, it likely needs to be zipped.

**Reply by Varunda** [reply]
Correct, the input is not from Imgur, it is a file on my computer. I have posted a tga used above

**Reply by fmw42** [reply]
You have provided a DDS (Microsoft DirectDraw Surface) file not a TGA file. Please post your TGA file

**Reply by Varunda** [reply]
Sorry about that, the tga is created from the DDS. I've attached a tga
[Indar_-16_12.0_color_tga.zip](https://github.com/ImageMagick/ImageMagick/files/6161485/Indar_-16_12.0_color_tga.zip)

**Reply by fmw42** [reply]
When I look at your tga file, it is bright. When I convert it to PNG, it keeps the same brightness. Perhaps the issue is your viewer and profiles. Here is the PNG, that I get, which is the same as the tga.

![Indar_-16_12 0_color](https://user-images.githubusercontent.com/13458218/111651857-d5c8a100-87c3-11eb-920b-8d414413a946.png)

`magick compare -metric rmse Indar_-16_12.0_color.tga Indar_-16_12.0_color.png null:`
0 (0)

A compare as per above, shows no difference.

**Comment by snibgo** [comment]
@Varunda: Indar_-16_12.0_color.tga has an alpha channel, with an average transparency of about 50%. "magick montage" removes transparency (EDIT: by flattening against the "-background" colour). So beware when comparing images that you may be comparing apples and oranges.

```
magick montage Indar_-16_12.0_color.tga -duplicate 3 x0.png
```
![x0](https://user-images.githubusercontent.com/33812368/111664906-2c58ce00-880a-11eb-8fc8-65d4e1427849.png)

```
magick montage Indar_-16_12.0_color.tga -alpha off -duplicate 3 x1.png
```
![x1](https://user-images.githubusercontent.com/33812368/111664935-31b61880-880a-11eb-84e6-52cfa11e4593.png)

```
magick montage Indar_-16_12.0_color.tga -background Red -duplicate 3 x2.png
```
![x2](https://user-images.githubusercontent.com/33812368/111664963-38dd2680-880a-11eb-9cd6-c2c1f93bae42.png)

**Reply by Varunda** [reply]
Using that flag fixes the issue. Thank you for the help!

**Comment by Varunda** [comment, marked as answer]
Is there a way I can mark a discussion as solved/closed? I'm not seeing a button to do that

**Comment by snibgo** [comment]
You can choose a reply and mark it as the "answer".
